newIDE/app/src/UI/SemiControlledAutoComplete.js

```diff
@@ -137,221 +65,192 @@ type State = {|
+            {option.renderIcon && (
+              <ListItemIcon>{option.renderIcon()}</ListItemIcon>
+            )}
+            {option.value}
+          </ListItem>
+        );
+      }
+    };
 
-                    // Style:
-                    style: props.textFieldStyle,
-                    fullWidth: props.fullWidth,
-                    ...computeTextFieldStyleProps(props),
+    const placeholder = () => {
+      if (this.props.errorText) return this.props.errorText.props.id;
+      else if (this.props.hintText && typeof this.props.hintText === 'string')
+        return this.props.hintText;
+      else return null;
+    };
 
-                    inputRef: this._input,
-                  })}
-                  <Popper
-                    open={isOpen}
-                    anchorEl={this._input.current}
-                    style={styles.menuPopper}
-                  >
-                    <div
-                      {...(isOpen
-                        ? getMenuProps({}, { suppressRefError: true })
-                        : {})}
-                    >
-                      {isOpen ? (
-                        <Paper
-                          style={{
-                            ...styles.menuPaper,
-                            width: this._input.current
-                              ? this._input.current.clientWidth
-                              : undefined,
-                          }}
-                          square
-                        >
-                          {filterDataSource(props.dataSource, inputValue).map(
-                            (item, index) =>
-                              renderItem({
-                                item,
-                                index,
-                                menuItemProps: getItemProps({ item, index }),
-                                highlightedIndex,
-                                selected: item === selectedItem,
-                              })
-                          )}
-                        </Paper>
-                      ) : null}
-                    </div>
-                  </Popper>
-                </div>
-              );
-            }}
-          </Downshift>
-        )}
-      </I18n>
+    return (
+      <Autocomplete
+        freeSolo
+        clearOnEscape
+        disableOpenOnFocus={this.props.openOnFocus}
+        onChange={handleChange}
+        style={{
+          ...styles.container,
+        }}
+        inputValue={currentInputValue()}
+        onInputChange={handleInputChange}
+        options={this.props.dataSource}
+        renderOption={renderItem}
+        getOptionDisabled={this.disableDivider}
+        ListboxProps={{
+          style: {
+            padding: 0,
+            margin: 0,
+          },
+        }}
+        getOptionLabel={handleOptionLabel}
+        filterOptions={filterFunction}
+        renderInput={params => {
+          nullifyDefaultStyling(params);
```

Comment:
While this works, you're mutating objects. Prefer to create new in this case:

const defaultStylingProps = getDefaultStylingProps(params);